Feat/product 리팩토링 및 소프트딜리트 개선작업

yjn33 opened the pull request:
## 📌 PR 요약
- ProductResponseDto.java에 클래스변수 User user를 UserInfoResponseDto user로 대체
- Product.java Entity에 @Setter 제거 리팩토링
- 소프트딜리트 방식 개선 작업

## 🔗 관련 이슈
- 관련된 이슈 번호를 연결해주세요.
    - 해당 없음

## 🛠️ 변경 사항
- 주요 변경 사항을 bullet으로 작성해주세요.
- api/v1/products

## 📸 스크린샷 (선택)

![스크린샷 2025-04-17 20-36-24](https://github.com/user-attachments/assets/f7c99155-7bc4-48f5-9810-780f8f89f7c7)
User 데이터 응답을 수정하였습니다

![스크린샷 2025-04-18 10-34-05](https://github.com/user-attachments/assets/29f90c3e-c7fc-4847-bf5e-f84d4c97192f)
현재 저장된 상품 게시물입니다.

![스크린샷 2025-04-18 10-32-23](https://github.com/user-attachments/assets/551e7c80-01ec-4e27-849c-b85f3059d468)
상품게시물 다건 조회시 1, 3번 상품만 조회되어지고, 2번상품은 조회가 안됩니다.

![스크린샷 2025-04-18 10-32-42](https://github.com/user-attachments/assets/188244f1-3cc7-449a-9869-386fcd990643)
![스크린샷 2025-04-18 10-33-02](https://github.com/user-attachments/assets/8e83b104-b8b2-4fc6-8d45-bd9e8a321a6b)
![스크린샷 2025-04-18 10-33-07](https://github.com/user-attachments/assets/81ce787f-7c33-4962-baa2-a51beb1efd9c)
마찬가지로 단건 조회시 2번 상품은 소프트딜리트 상태이기때문에 정상적으로 조회가 안됩니다.



## ⚠️ 주의 사항 (선택)
리뷰어가 알아야 할 중요한 사항이나 주의할 점을 작성해주세요.


## ✅ 체크리스트
PR 작성자가 확인해야 할 항목입니다:
- [x] 코드가 정상적으로 동작하는지 확인했습니다.
- [x] 관련 문서를 업데이트했습니다.
- [x] 코드 리뷰어를 등록했습니다.
- [x] Labels를 등록했습니다.

yjn33 committed:
Refactor: 상품 응답시 user정보 ResponseDto 리팩토링

User user -> UserInfoResponseDto user 로 리팩토링

UserInfoResponseDto.j
 -> fromProduct 정적 팩토리 매서드 적용

yjn33 committed:
Refactor: Product @Setter 제거

yjn33 committed:
Merge branch 'develop' into feat/product

yjn33 commented:
softdelete 부분 이슈로 pr 임시 폐쇄

yjn33 committed:
Perf: 상품 게시물 소프트 딜리트 개선 작업

deletedAt의 null값을 조회하는것이 아닌 ProductStatus의 DELETED를 기준으로 조회

emily101304 commented on src/main/java/com/example/junggoheaven/domain/product/service/component/ProductFinder.java:
```diff
@@ -21,12 +21,13 @@ public class ProductFinder {
 	public Page<Product> findAllProduct(Pageable pageable) {
-		return productRepository.findByDeletedAtNull(pageable);
+		return productRepository.findAllAndStatusIsNotDeleted(pageable);
+		//return productRepository.findByDeletedAtNull(pageable);
```
이 주석처리는 계속 사용하시는 건가요~?

yjn33 replied:
코드 수정 부분 표시할려고 남겨둔 것입니다 최종배포시에 주석은 다 제거할 예정입니다.

yjn33 replied:
커밋으로 확인이 가능하니 제거하겠습니다

JeisaNewbie commented on src/main/java/com/example/junggoheaven/domain/product/repository/ProductRepository.java:
```diff
@@ -1,16 +1,25 @@
+	Optional<Product> findByIdAndStatusIsNotDeleted(Long productId);
+
+	@Query("SELECT i FROM Product i WHERE i.status <> 'DELETED'")
+	Page<Product> findAllAndStatusIsNotDeleted(Pageable pageable);
```
Product Entity 에 @Filter(name = "deletedFilter", condition = "status <> 'DELETED'") 가 적용돼있으면 
WHERE i.status <> 'DELETED' 가 필요 없을거같습니다.

JeisaNewbie replied:
따라서 메서드 에도 AndStatusIsNotDeleted 를 추가 할 필요는 없어보이네요.

yjn33 replied:
확인중입니다

yjn33 replied:
https://github.com/JungGo-Heaven/JungGo-Heaven/pull/34/commits/75861f739a2c973be31a409bd5400604d2109aec
여기 커밋 참고해주세요 FilterAespect.java에서 enableFilter 메서드 aop repository 경로 수정하고 ProductFinder.java의 findAllProduct 매서드 에서productRepository.findAllAndStatusIsNotDeleted(pageable); 대신productRepository.findAll(pageable);를 실행하니 @Query 사용하지 않아도 product 엔터티에 있는 @Filter(name = "deletedFilter", condition = "status <> 'DELETED'") 가 소프트딜리트 상품 잘 걸러주네요

yjn33 replied:
추가 버그 확인하고 수정된 커밋 올리겠습니다

yjn33 replied:
push 했습니다 참고바랍니다

yjn33 committed:
fix: filter 실행 경로범위 변경

경로를 수정하여 모든 repository필터를 찾을수 있게 수정

yjn33 committed:
fix: 소프트 딜리트 상품 조회시 @Query 개선작업

상품 다건 조회시 Product entity의 @Filter가 소프트 딜리트 상품 필터링 해줌으로, 불필요한 @Query Annotation은 제거
